OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -164,7 +166,7 @@ public Actor FindBaseProvider(World world, Player p, CPos topLeft)
 
 				// Range is counted from the center of the actor, not from each cell.
 				var target = Target.FromPos(bp.Actor.CenterPosition);
-				if (target.IsInRange(center, bp.Trait.Info.Range))
+				if (target.IsInRange(center, bp.Trait.Info.Range) || !buildRadiusEnabled)
```

Comment:
Please change this to return null before the loop if `!buildRadiusEnabled`, and then:

1. Query `MapBuildRadius` in the `Building` constructor and store the result in a `readonly bool buildRadiusEnabled`.
2. Check that value in `IsCloseEnoughToBase` to avoid calling this method entirely.